i2cdisplaybus/__init__.py

```diff
@@ -87,7 +87,9 @@ def send(self, command: int, data: ReadableBuffer) -> None:
         done.
         """
         self._begin_transaction()
-        self._send(DISPLAY_COMMAND, CHIP_SELECT_UNTOUCHED, bytes([command] + data))
+        # re-wrap in case of byte-string
+        buffer = list(data) if isinstance(data, bytes) else data
+        self._send(DISPLAY_COMMAND, CHIP_SELECT_UNTOUCHED, bytes([command] + buffer))
```

Comment:
This is going to be super slow. Maybe instead `_send` could take an optional `command` argument, and add it to `command_bytes` before the rest of the data if it's specified?

Reply:
There are **very** few instances where a binary-character literal is actually used, since most command/buffer combos are described using real numeric arrays, thus only occurring the overhead of the `isinstance` check.

Perhaps the more interesting question **should** be is why did my change on the SSD1306 driver _break_ non-Blinka devices? They should already be able to take numeric-arrays since all the other command-buffer combinations work, what's so special about an empty array vs an empty binary-character literal?

Reply:
I agree with @deshipu, just change `_send()` to take command directly and skip the extra conversion.

Note that data should *not* be a list. It should always be a bytes, bytearray or array.

Reply:
Ok, it took a bit of back-tracking to see what y'all were referring to (the usage in `_displaycore` vs `i2cdisplaybus`). Changes forthcoming...

Reply:
Instead of creating a bytearray in `send`, add an argument to `_send()`. `_send()` is allocating a bytearray as well that it can put command into. That way you'll only make one new bytearray, not two.

Reply:
Note that the _original_ code **also** "created a new buffer" with `bytes([command] + buffer)`, so this change is not actually adding any overhead and is much simpler and easier to understand, and avoids issues caused by the OG code.

It also avoids **any** changes to method signatures and buffer construction _elsewhere_.

Reply:
Er, I tend to use _empahsis_ when I comment and such, so it's not to be taken **personally** :smiley:

Reply:
> I agree with @deshipu, just change `_send()` to take command directly and skip the extra conversion.
> 
> Note that data should _not_ be a list. It should always be a bytes, bytearray or array.

Not to be (too overly) pedantic, but I think that's part of the initial issue -- `bytes` don't work:

```python3
>>> from os import urandom
>>> data = urandom(8)
>>> type(data)
<class 'bytes'>
>>> cmd = 0x80
>>> type(cmd)
<class 'int'>
>>> [cmd] + data
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate list (not "bytes") to list

```

Reply:
What happens when you send it in 2 calls like in [FourWire](https://github.com/adafruit/Adafruit_Blinka_Displayio/blob/main/fourwire/__init__.py)?

Reply:
Just walking through this again, the difference, I believe, is that I2C uses buffered writes, while it appears the FourWire is just jamming byte-at-a-time.

If it's changed to two `_send` commands (within the same transaction):
```
# create an array to create a `ReadableBuffer`
self._send(DISPLAY_COMMAND, CHIP_SELECT_UNTOUCHED, [command])
# create buffer and write
self._send(DISPLAY_COMMAND, CHIP_SELECT_UNTOUCHED, data)
# create buffer and write
```
That appears to require a lot more buffer creation, so even more overhead.

My surmise is that this is because I2C is an _addressable_ format, thus easier to lock and send a bunchastuff at once, while FourWire is chip-select and can tolerate interleaving of bytes. 

(Sorry if I'm getting windy, but I recently retired from software development and this the best conversation I've had along those lines in _months_!)

Reply:
> It also avoids **any** changes to method signatures and buffer construction _elsewhere_.

`_send()` is a private function (designated by the `_`) used by just this function. It's not a big deal to change its method signature. So, please add command to the signature and then pack it into a buffer inside `_send()`.

Reply:
Ok, I'm not comfortable with that change, thanks.